### PR TITLE
fix: NothingPersonalValidator strip_explode too sensible

### LINKED ISSUE #385: Bug: NothingPersonalValidator strip_explode too sensible
### PHP Version

7.4

### CodeIgniter4 Version

4.2.4

### Shield Version

1.0.0-beta.2

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### Database

MySQL 5.6

### Did you customize Shield?

No

### What happened?

The function NothingPersonalValidator->strip_explode() is too sensible for exploding the user password.
Indeed, it explode any part separated for example with an underscore or a "-".
This can produce the search of only one letter in the username and/or email adresse as part of a personal information.
The user is then prompted wrongly that his password cannot contain any personal information contained is his username or email.

### Steps to Reproduce

Try to register with : 
email : xxx@gmail.com
password : G-test#1234

In this case, NothingPersonalValidator will fail cause "G" is considered as part of "xxx@gmail.com"

### Expected Output

I think the NothingPersonalValidator->strip_explode() function should check if the searched string is not too short and then try to make a match only if the searching string contain 3 or more caracters.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by arif-rh**
This could fixes #385  and closes #389

### CODE REVIEW

**Comment by datamweb** [comment]
Hello. Please fix coding style.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#php-style

File: src/Authentication/Passwords/NothingPersonalValidator.php [hunk 1]
```diff
@@ -114,12 +114,12 @@ protected function isNotPersonal(string $password, ?User $user): bool
             $haystacks = $this->strip_explode($password);
 
             foreach ($haystacks as $haystack) {
-                if (empty($haystack) || in_array($haystack, $trivial, true)) {
+                if (empty($haystack) || in_array($haystack, $trivial, true) || mb_strlen($haystack, 'UTF-8') < 3) {
                     continue;  //ignore trivial words
```
Comment (kenjis) [inline src/Authentication/Passwords/NothingPersonalValidator.php:118]:
```suggestion
                    continue;  // ignore trivial words
```

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
LGTM.

**Review by MGatner — APPROVED** [review]
Thanks @arif-rh!

**Pull request merged**